File: src/MyApplication.Web.UI/src/api/authApi.ts
import { apiRequest } from "./client";
import type { AuthResponse } from "./types";

const BASE_PATH = "/Auth";

export function loginWithGoogle(idToken: string): Promise<AuthResponse> {
  return apiRequest(`${BASE_PATH}/google`, { method: "POST", body: JSON.stringify({ idToken }) });
}

export function refresh(): Promise<AuthResponse> {
  return apiRequest(`${BASE_PATH}/refresh`, { method: "POST" });
}

export function logout(): Promise<void> {
  return apiRequest(`${BASE_PATH}/logout`, { method: "POST" });
}

File: src/MyApplication.Android/app/src/main/kotlin/io/github/rualert/mynotesapp/data/api/AuthApi.kt
package io.github.rualert.mynotesapp.data.api

import retrofit2.http.Body
import retrofit2.http.POST

/**
 * Точка входа в сессию. Refresh-token в теле не ходит — он живёт в cookie,
 * которую хранит [io.github.rualert.mynotesapp.data.auth.SessionCookieJar].
 */
interface AuthApi {

    @POST("Auth/google")
    suspend fun loginWithGoogle(@Body request: GoogleLoginRequest): AuthResponse

    @POST("Auth/refresh")
    suspend fun refresh(): AuthResponse

    @POST("Auth/logout")
    suspend fun logout()
}
